issue16161.py

```diff
@@ -0,0 +1,44 @@
+from sympy import *
+init_printing()
+from functools import reduce 
+from sympy.abc import x
+def by_parts_integration(eq, args = []):
+
+    diffr = [ ]
+    integrates = [ ] 
+
+    #Improve this
+    if len(factor_list(eq)[1])< 2:
+        if args:
+            return integrate(eq, args)
+        else:
+            return integrate(eq)
+
+    for i,j in factor_list(eq)[1]:
+        if type(i) in [type(sec(x)), type(cos(x)), type(sin(x)), type(tan(x)), 
+                       type(csc(x)), type((cot(x))), type(exp(x))]:        
+            integrates.append((i, j))
+
+        else:    
+#         if type(i) in ["<class 'sympy.core.power.Pow'>", 

```

Comment:
For maintaining consistency in the codebase, you can use [isinstance](https://www.programiz.com/python-programming/methods/built-in/isinstance) for such checks.

Reply:
Ok.